dali/pipeline/data/tensor.h

```diff
@@ -428,16 +428,44 @@ class Tensor : public Buffer<Backend> {
   }
 
   /**
-   * @brief Remove single-dimensional entries from the shape
-   * of a Tensor
+   * @brief Remove any single-dimensional entries from the shape
+   * of a Tensor.
    */
   inline void Squeeze() {
-    std::vector<Index> shape(shape_.begin(), shape_.end());
-    shape.erase(std::remove(shape.begin(), shape.end(), 1), shape.end());
-    if (shape.empty()) {
-      shape.push_back(1);
+    SmallVector<int64_t, 6> out_shape;
+    TensorLayout out_layout;
+    TensorLayout in_layout = GetLayout();
+    for (int d = 0; d < shape_.size(); d++) {
+      if (shape_[d] == 1)
+        continue;
+      out_shape.push_back(shape_[d]);
+      if (!in_layout.empty())
+        out_layout += in_layout[d];
+    }
+    shape_ = out_shape;
+    SetLayout(out_layout);
+  }
+
+  /**
+   * @brief Removes the specified dimension from the shape, if its extent is
+   * equal to 1.
+   * @param dim Dimension to be squeezed. Negative indexing is also suppo9rted
+   */
+  inline void Squeeze(int dim) {
```

Comment:
Following that definition (the function does or does not squeeze), maybe return a bool?
```suggestion
  inline bool Squeeze(int dim) {
```